pkg/build/api/types.go

```diff
@@ -570,6 +570,25 @@ type CustomBuildStrategy struct {
 	BuildAPIVersion string
 }
 
+// ImageOptimizationPolicy describes what optimizations the builder can perform when building images.
+type ImageOptimizationPolicy string
+
+const (
+	// ImageOptimizationNone will attempt to generate canonical Docker image.
+	ImageOptimizationNone ImageOptimizationPolicy = "None"
+
+	// ImageOptimizationSkipLayerCache is an experimental policy and will avoid creating
+	// unique layers for each dockerfile line, resulting in smaller images and saving time
+	// during creation. Some Dockerfile syntax is not fully supported - content added to
+	// a VOLUME by an earlier layer may have incorrect uid, gid, and filesystem permissions.
+	// If an unsupported setting is detected, the build will fail.
```

Comment:
this seems to promise more than we can deliver in terms of compatibility.

i feel like it needs some additional wording about the fact that there may be other incompatibilities (especially if docker changes their behavior in the future) that we may not detect/warn about.